python/tank/platform/interactive_login.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+UI and console based login for Toolkit.
+"""
+
+from getpass import getpass
+import logging
+
+from tank_vendor.shotgun_api3 import Shotgun, AuthenticationFault
+from tank.util.session import cache_session_data, get_cached_login_info, delete_session_data, is_session_token_cached
+from tank.util.login import get_login_name
+from tank.util.shotgun import get_associated_sg_config_data, get_associated_sg_base_url
+
+# Configure logging
+logger = logging.getLogger("sgtk-interactive_login")
+# logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+
+def _get_session_token(hostname, login, password, http_proxy):
+    """
+    Retrieves a session token for the given credentials.
+    :param hostname: The host to connect to.
+    :param login: The user to get a session for.
+    :param password: Password for the user.
+    :param http_proxy: Proxy to use. Can be None.
+    :returns: If the credentials were valid, returns a session token, otherwise returns None.
+    """
+    try:
+        # Create the instance...
+        sg = Shotgun(
```

Comment:
Looks like we now have three places in the code where we instantiate a `Shotgun()` instance. Before we only had one place in the main body of code. Wondering if this is bad? Also, it seems the new methods utf-8 encode the connection details where the old one doesn't. I think it should be consistent. Why do we need to do this encoding? Are we expecting unicode objects? The rest of the core API assumes that all items are strings (try grepping for encode!). I think this is a good rule generally. Often QT returns unicode rather than strings, in that case we generally tend to convert at the QT level rather than the core level. Does that make sense in this case you think?